pkg/auth/authorizer/abac/abac.go

```diff
@@ -100,13 +101,20 @@ func NewFromFile(path string) (policyList, error) {
 func (p policy) matches(a authorizer.Attributes) bool {
 	if p.subjectMatches(a) {
 		if p.Readonly == false || (p.Readonly == a.IsReadOnly()) {
-			if p.Resource == "" || (p.Resource == a.GetResource()) {
+			switch {
+			case p.NonResourcePath != "":
+				if p.NonResourcePath == a.GetNonResourcePath() {
+					return true
+				}
+			// When the path is a non-resource path it cannot match.
+			case len(a.GetNonResourcePath()) == 0 && (p.Resource == "" || (p.Resource == a.GetResource())):
```

Comment:
@liggitt: `case len(a.GetResource()) > 0 && ...` would be a breaking change, as well. At least it breaks the tests: `{"user":"alice","namespace":"projectCaribou"}` should be allowed to a ns w/o a resource (`{User: uAlice, RO: true, Resource: "", NS: "projectCaribou", ExpectAllow: true},`)